src/test/auxiliary/derive-no-std.rs

```diff
@@ -8,32 +8,33 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![feature(rand, collections, rustc_private)]
-#![no_std]
+// no-prefer-dynamic
```

Comment:
Actually `#![crate_type = "rlib"]` _and_ `// no-prefer-dynamic` are required for it not to try and resolve the lang items.